Classes/Service/HtmlMinifier.php

```diff
@@ -103,4 +103,22 @@ protected function isFeatureActive(string $feature): bool
     {
         return isset($GLOBALS['TYPO3_CONF_VARS']['EXTENSIONS']['minify'][$feature]) && $GLOBALS['TYPO3_CONF_VARS']['EXTENSIONS']['minify'][$feature] === '1';
     }
+
+    private function preserveTypo3Comment(string $html): string
+    {
+        $typo3CommentStart = strpos($html, '<!--');
+        if ($typo3CommentStart !== false) {
+            $typo3CommentStop = strpos($html, '-->', $typo3CommentStart);
+            $typo3Comment = substr($html, $typo3CommentStart, $typo3CommentStop - $typo3CommentStart + 3);
+
+            if (strpos($typo3Comment, 'TYPO3') !== false) {
+                return $typo3Comment;
+            } else {
+                $html = str_replace($typo3Comment, '', $html);
+                return $this->preserveTypo3Comment($html);
```

Comment:
doesn't that remove comments such as these: <!--[if lte IE 8]>

Reply:
or `<!--TYPO3SEARCH_begin-->` `<!--TYPO3SEARCH_end-->`

Reply:
i guess that's the sense of removing comments
but you are warmly invited to commit your ideas in a seperate bugfix branch